programs/relayer/src/modules/eth_to_cosmos.rs

```diff
@@ -200,22 +197,12 @@ impl ModuleServer for EthToCosmosRelayerModule {
     }
 
     #[tracing::instrument(skip_all)]
-    async fn serve(
-        &self,
-        config: serde_json::Value,
-        addr: SocketAddr,
-    ) -> Result<(), tonic::transport::Error> {
+    async fn serve(&self, config: serde_json::Value) -> anyhow::Result<Box<dyn RelayerService>> {
```

Comment:
I think this is a confusing artifact that doesn't really make sense anymore. I think a more reasonable approach might be simply let the modules not need to care about _how_ the config gets to them, and instead just have the router that do this work and call new. I will look into this after I'm done with my current deployment work.